### PR TITLE
Fixes #1981 Converted SettingsPresenter to Kotlin

### LINKED ISSUE #1981: Convert SettingsPresenter to Kotlin
**Is your feature request related to a problem? Please describe.**

SettingsPresenter is still in Java. It can be converted to Kotlin.

**Describe the solution you'd like**

Convert it to kotlin

### PR DESCRIPTION
**Pull request by akkywadhwa**
Fixes #1981

**Screenshots** 
N/A

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/SettingsPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.kiwix.kiwixmobile.core.data.DataSource
+import org.kiwix.kiwixmobile.core.settings.SettingsContract.Presenter
+import org.kiwix.kiwixmobile.core.settings.SettingsContract.View
+import javax.inject.Inject
+
+internal class SettingsPresenter @Inject constructor(private val dataSource: DataSource) :
+  BasePresenter<View?>(), Presenter {
+  override fun clearHistory() {
+    dataSource.clearHistory()
+      .subscribe(object : CompletableObserver {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/SettingsPresenter.kt]: Do me a favour and get rid of this CompletableObserver. There is a subscribe function that takes 2 lambdas, one that gets invoked on completing of the completable and one that gets invoked if there is an error

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/SettingsPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        override fun onSubscribe(d: Disposable) {
+          // TODO
+        }
+
+        override fun onComplete() {
+          // TODO
+        }
+
+        override fun onError(e: Throwable) {
+          Log.e("SettingsPresenter", "$e")
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/SettingsPresenter.kt]: Fix this Log up too. Log has a 3 param version specifically for exceptions

**Comment by akkywadhwa** [comment]
@macgills anything else remaining in this?

**Comment by macgills** [comment]
All good! Was just waiting for a green build and then got distracted

**Pull request merged**